### PR TITLE
chore: Enable ESLint for apps and fix all errors

### PR DESCRIPTION
**Pull request by susnux**
## Summary

Nevertheless this causes a huge amount of new warnings. Previously the shell script for directories to lint was wrong it was generating all app names to lint, but was missing the `apps/` prefix. Causing only `core` to be linted.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by st3iny — APPROVED** [review]
LGTM! There are npm build changes though.

File: apps/federatedfilesharing/src/components/AdminSettings.vue [hunk 1]
```diff
@@ -54,7 +54,6 @@ import { showError } from '@nextcloud/dialogs'
 import axios from '@nextcloud/axios'
 import { generateOcsUrl } from '@nextcloud/router'
 import { confirmPassword } from '@nextcloud/password-confirmation'
-import '@nextcloud/password-confirmation/dist/style.css'
```
Comment (st3iny) [inline apps/federatedfilesharing/src/components/AdminSettings.vue:57]: I think this line should stay.

File: apps/files/src/actions/deleteAction.ts [hunk 2]
```diff
@@ -126,7 +126,7 @@ export const action = new FileAction({
 			.every(permission => (permission & Permission.DELETE) !== 0)
 	},
 
-	async exec(node: Node, view: View, dir: string) {
+	async exec(node: Node) {
```
Comment (nickvergessen) [inline apps/files/src/actions/deleteAction.ts:129]: I guess all actions implement the same methods with the same arguments, being them used or not.
So while unused, it kind of can help to know which others also exist? :see_no_evil:

File: apps/files/src/main.ts [hunk 3]
```diff
@@ -19,9 +19,9 @@ __webpack_nonce__ = btoa(getRequestToken())
 
 declare global {
 	interface Window {
-		OC: any;
-		OCA: any;
-		OCP: any;
+		OC: Nextcloud.v28.OC;
```
Comment (nickvergessen) [inline apps/files/src/main.ts:22]: Should this be 30 on master?

File: apps/files_sharing/src/services/SharingService.ts [hunk 4]
```diff
@@ -71,7 +73,7 @@ const ocsEntryToNode = async function(ocsEntry: any): Promise<Folder | File | nu
 				'owner-id': ocsEntry?.uid_owner,
 				'owner-display-name': ocsEntry?.displayname_owner,
 				'share-types': ocsEntry?.share_type,
-				favorite: ocsEntry?.tags?.includes(window.OC.TAG_FAVORITE) ? 1 : 0,
+				favorite: ocsEntry?.tags?.includes((window.OC as Nextcloud.v28.OC & { TAG_FAVORITE: string }).TAG_FAVORITE) ? 1 : 0,
```
Comment (nickvergessen) [inline apps/files_sharing/src/services/SharingService.ts:76]: Same here? v30 for master?

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue [hunk 5]
```diff
@@ -70,7 +70,6 @@ import moment from '@nextcloud/moment'
 import axios from '@nextcloud/axios'
 import { generateOcsUrl } from '@nextcloud/router'
 import { confirmPassword } from '@nextcloud/password-confirmation'
-import '@nextcloud/password-confirmation/dist/style.css'
```
Comment (nickvergessen) [inline apps/settings/src/components/BasicSettings/BackgroundJob.vue:73]: is it added globally somewhere or still needed here?

File: apps/settings/src/components/Encryption.vue [hunk 6]
```diff
@@ -68,7 +68,6 @@ import { getLoggerBuilder } from '@nextcloud/logger'
 
 import { generateOcsUrl } from '@nextcloud/router'
 import { confirmPassword } from '@nextcloud/password-confirmation'
-import '@nextcloud/password-confirmation/dist/style.css'
```
Comment (nickvergessen) [inline apps/settings/src/components/Encryption.vue:71]: same

File: apps/settings/src/components/WebAuthn/Section.vue [hunk 7]
```diff
@@ -38,7 +38,6 @@
 import { browserSupportsWebAuthn } from '@simplewebauthn/browser'
 import { confirmPassword } from '@nextcloud/password-confirmation'
 import NcNoteCard from '@nextcloud/vue/dist/Components/NcNoteCard.js'
-import '@nextcloud/password-confirmation/dist/style.css'
```
Comment (nickvergessen) [inline apps/settings/src/components/WebAuthn/Section.vue:41]: same

File: apps/sharebymail/src/components/AdminSettings.vue [hunk 8]
```diff
@@ -28,7 +28,6 @@ import { showError } from '@nextcloud/dialogs'
 import axios from '@nextcloud/axios'
 import { generateOcsUrl } from '@nextcloud/router'
 import { confirmPassword } from '@nextcloud/password-confirmation'
-import '@nextcloud/password-confirmation/dist/style.css'
```
Comment (nickvergessen) [inline apps/sharebymail/src/components/AdminSettings.vue:31]: same

File: apps/twofactor_backupcodes/src/views/PersonalSettings.vue [hunk 9]
```diff
@@ -49,7 +49,6 @@
 
 <script>
 import { confirmPassword } from '@nextcloud/password-confirmation'
-import '@nextcloud/password-confirmation/dist/style.css'
```
Comment (nickvergessen) [inline apps/twofactor_backupcodes/src/views/PersonalSettings.vue:52]: same

File: apps/files/src/actions/deleteAction.ts (hunk 2 above)
Comment (susnux) [reply apps/files/src/actions/deleteAction.ts:129]: For this we have the type interface (+ your IDE should be able to show you all available parameters).
Having unused parameters is considered an error by our code style.

Comment (susnux) [reply apps/files/src/actions/deleteAction.ts:129]: *(I think this was the only action that had unused parameters still be defined all other only take what they use)*

File: apps/files/src/main.ts (hunk 3 above)
Comment (susnux) [reply apps/files/src/main.ts:22]: Not available yet. We do not even have types for v29, release still waits for approval:
https://github.com/nextcloud-libraries/nextcloud-typings/pull/260

File: apps/files_sharing/src/services/SharingService.ts (hunk 4 above)
Comment (susnux) [reply apps/files_sharing/src/services/SharingService.ts:76]: See other comment, not available yet

File: apps/federatedfilesharing/src/components/AdminSettings.vue (hunk 1 above)
Comment (susnux) [reply apps/federatedfilesharing/src/components/AdminSettings.vue:57]: You are right, reason for ESLint to remove this is a typo in the password-confirmation package...

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue (hunk 5 above)
Comment (susnux) [reply apps/settings/src/components/BasicSettings/BackgroundJob.vue:73]: We can consider doing so but currently no it is not

**Review by ShGKme — APPROVED** [review]
Thanks!

Checked all the changes, looks good.
I expected much more changes 👀

The only thing I'm working about is removing the import. Having a file it is not easily possible to say if removing this line is safe. And strictly, each file with dialogs must have dialog styles.
```js
import '@nextcloud/password-confirmation/dist/style.css'
```

**Review by ChristophWurst — APPROVED** [review]

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue (hunk 5 above)
Comment (nickvergessen) [reply apps/settings/src/components/BasicSettings/BackgroundJob.vue:73]: But it is then still needed?

Comment (susnux) [reply apps/settings/src/components/BasicSettings/BackgroundJob.vue:73]: Yes current it is

Comment (susnux) [reply apps/settings/src/components/BasicSettings/BackgroundJob.vue:73]: I rather think it makes sense to have this in the library as this should only be added if needed, but currently we need it. (fixed now)

**Comment by susnux** [comment]
Current blocker:
Some types are missing from our typings of OC / OCP, I created a PR so waiting for this and then a new release:
- [x] https://github.com/nextcloud-libraries/nextcloud-typings/pull/263
- [ ] https://github.com/nextcloud-libraries/nextcloud-typings/pull/264
- [ ] https://github.com/nextcloud-libraries/nextcloud-typings/pull/262

File: apps/files/src/main.ts (hunk 3 above)
Comment (susnux) [reply apps/files/src/main.ts]: At least it is 29 now...

See my proposal for modernizing it which also will add a "latest" alias:
https://github.com/nextcloud-libraries/nextcloud-typings/pull/261

**Comment by skjnldsv** [comment]
Can we do that just after branch off 31 please :pray:

**Pull request merged**